src/hooks/pokemonDataHooks.tsx

```diff
@@ -0,0 +1,98 @@
+import { useQuery, useInfiniteQuery } from "react-query";
+import { pokemonService } from "../services/pokemon.service";
+import { Pokemon, PokemonQueryParams } from "../interfaces/Pokemon";
+
+export const usePokemonTableData = ({
+  page,
+  rowsPerPage,
+  sort_by,
+  sort_order,
+  searchValue,
+  selectedCtg,
+  userId,
+}: PokemonQueryParams) => {
+  const queryResult = useQuery(
+    [
+      "pokemonData",
+      page,
+      rowsPerPage,
+      sort_by,
+      sort_order,
+      searchValue,
+      selectedCtg,
+    ],
+    () =>
+      pokemonService.getPokemons(
+        selectedCtg === 1 ? userId : undefined,
+        page!,
+        rowsPerPage!,
+        sort_by,
+        sort_order,
+        searchValue
+      ),
+    { refetchOnWindowFocus: false, keepPreviousData: true }
+  );
+
+  const { data: userPokemonData } = useUserPokemons(userId);
+
+  const pokemonsWithOwnership =
+    queryResult.data?.pokemons.map((pokemon: Pokemon) => ({
+      ...pokemon,
+      belongsToUser:
+        userPokemonData?.pokemons.some(
+          (userPokemon: Pokemon) => userPokemon.id === pokemon.id
+        ) || false,
+    })) || [];
+
+  return {
+    ...queryResult,
+    data: {
+      ...queryResult.data,
+      pokemons: pokemonsWithOwnership,
+    },
+  };
+};
+
+export const usePokemonGridData = ({
+  sort_by,
+  sort_order,
+  searchValue,
+  selectedCtg,
+}: PokemonQueryParams) => {
+  return useInfiniteQuery(
+    ["infinitePokemonData", selectedCtg, sort_by, sort_order, searchValue],
+    ({ pageParam = 0 }) =>
+      pokemonService.getPokemons(
+        selectedCtg === 1 ? "02fea148-e9dc-4cae-89aa-8db50df0dd48" : undefined,
```

Comment:
is this the hardcoded id?

Reply:
About the hardcoded ID, it was to prepare for authentication. After merging, I will need to integrate the user with the service that fetches the data using the ID of the logged-in user.